darwin/options.py

```diff
@@ -528,6 +532,9 @@ def parse_args(self) -> Tuple[Namespace, ArgumentParser]:
         """
         args = self.parser.parse_args()
 
+        if args.action == "pull" and args.folders is not True:
+            args.folders = args.folders.lower() == "true"
+
```

Comment:
does this have any problems since `args.folders` is a bool?

Reply:
Apologies, I should have added a comment here before. This isn't clean but it's necessary to preserve the optional nature of the `--folders` flag. Currently, `--folders` behaves like this:
- If not passed, the param is `False` (pull flat structure)
- If passed, the param is `True` (pull folder structure)
- It does not accept any passed value

To preserve as much existing API behaviour as possible, the goal is this PR is the following:
- If not passed, the param is `True` (pull folder structure)
- If passed, the param is `True` (pull folder structure)
- If passed with a value, unless that value is `True`, set the param to `False` (Essentially still allowing CLI-based pulls to be pulled as flat structures)

Reply:
Re-writing to employ the use of a `--no-folders` arg